apps/src/util/exporter.js

```diff
@@ -73,9 +98,98 @@ export async function generateExpoApk(options, config) {
   } = appJson.expo;
   appJson.expo = onlineOnlyExpo;
 
-  const artifactUrl = await session.getApkUrlAsync(appJson);
+  if (buildMode) {
+    return expoBuildApk(session, appJson.expo);
+  } else {
+    return expoCheckApkBuild(session, appJson.expo, apkBuildId);
+  }
+}
+
+async function expoBuildApk(session, manifest) {
+  const result = await session.buildAsync(manifest, {
+    platform: 'android',
+    mode: 'create',
+    isSnack: true,
+    sdkVersion: EXPO_SDK_VERSION
+  });
+  const {id} = result;
+  return id;
+}
+
+async function expoCheckApkBuild(session, manifest, apkBuildId) {
+  const result = await session.buildAsync(manifest, {
+    platform: 'android',
+    mode: 'status',
+    current: false
+  });
+  const {jobs = []} = result;
+  const job = jobs.find(job => apkBuildId && job.id === apkBuildId);
+  if (!job) {
+    throw new Error(`Expo build not found: ${apkBuildId}`);
```

Comment:
Should we instead present the user with an error message?

Reply:
(same in line 135)

Reply:
(and line 191)

Reply:
Yes, throwing this error will result in the user being presented with an error message within `ExportDialog`

`exportError` or `apkError` will be set on its state (depending on which phase of the build resulted in an error). The render function will then say `Error creating Android Package`

Reply:
👍